### PR TITLE
Remove NVFUSER_DISTRIBUTED. 

### PR DESCRIPTION
**Pull request by wujingyue**
@xwang233 discussed this in the nvFuser-MultiGPU chatroom. At this moment, supporting non-distributed build of pytorch isn't worth the cost of additional CI and using knobs like `NVFUSER_DISTRIBUTED` or `USE_DISTRIBUTED`. 

This PR reverts https://github.com/NVIDIA/Fuser/pull/1711

Feel free to revert this PR if non-distributed build becomes important.

### CODE REVIEW

**Comment by wujingyue** [comment]
!build --dist

**Comment by samnordmann** [comment]
Can you please explain the motivation for this change?

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,13 +15,6 @@ set(NVFUSER_THIRD_PARTY_DIR "${NVFUSER_ROOT}/third_party")
 option(NVFUSER_STANDALONE_BUILD_WITH_UCC "" OFF)
 option(NVFUSER_BUILD_WITH_ASAN "Build nvFuser with asan" OFF)
 
-include(CMakeDependentOption)
-cmake_dependent_option(NVFUSER_DISTRIBUTED "" ON "USE_DISTRIBUTED" OFF)
```
Comment (samnordmann) [inline CMakeLists.txt:19]: don't we need to assert that `USE_DISTRIBUTED` in `ON`?

**Comment by wujingyue** [comment]
> Can you please explain the motivation for this change?

@xwang233 talked about this in the nvFuser-MultiGPU chatroom. At this moment, there isn't much value to support non-distributed build of pytorch. The cost of additional CI and knobs like NVFUSER_DISTRIBUTED/USE_DISTRIBUTED just for that config isn't worthwhile.

File: CMakeLists.txt (hunk 1 above)
Comment (wujingyue) [reply CMakeLists.txt:19]: Great point -- we should make the error show up early in config time not in build time.

File: setup.py [hunk 2]
```diff
@@ -306,16 +296,13 @@ def cmake(install_prefix: str = "./nvfuser"):
 
     logger.setLevel(logger_level)
 
-    pytorch_use_distributed = get_pytorch_use_distributed()
```
[the remark this answers is not in the document]
Comment (xwang233) [reply setup.py:309]: From nvfuser side, this is not officially supported. It may or may not work.

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build --dist

File: setup.py [hunk 3]
```diff
@@ -306,7 +299,9 @@ def cmake(install_prefix: str = "./nvfuser"):
...
+    assert (
```
Comment (wujingyue) [inline setup.py:302]: @xwang233 is this the right way to check a misconfig? I usually use `assert` only for programming errors, but I'm fine with using it if no alternatives.

File: CMakeLists.txt (hunk 1 above)
Comment (wujingyue) [reply CMakeLists.txt:19]: I added the assertion in setup.py. PTAL

File: setup.py (hunk 3 above)
Comment (xwang233) [reply setup.py:302]: I would personally prefer a `RuntimeError` clause like this one in apex https://github.com/NVIDIA/apex/blob/a7de60e57f0534266841e1733262601ad76aaa74/setup.py#L39-L47

Comment (wujingyue) [reply setup.py]: Done.

**Comment by wujingyue** [comment]
!build --dist

**Comment by wujingyue** [comment]
Glad that CI has recovered, making this PR ready to review again. PTAL

**Review by cowanmeg — APPROVED** [review]
LGTM for multidevice code and tests. I would wait for @xwang233's stamp on the build changes.

**Review by xwang233 — APPROVED** [review]
LGTM. Thanks for working on this.

**Pull request merged**